dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/EventController.java

```diff
@@ -885,4 +889,38 @@ protected boolean hasHref( List<String> fields )
         return metaData;
     }
 
+    private IdSchemes getIdschemes( IdSchemes idSchemes, Map<String, List<String>> params )
+    {
+        Field[] fields = idSchemes.getClass().getDeclaredFields();
```

Comment:
Is reflection really necessary here? there are not that many types of scheme available, probably would be better with a static check